src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -151,14 +153,24 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 				{
 					Constraint *constraint = (Constraint *) command->def;
 					char **constraintName = &(constraint->conname);
+					const bool missingOk = false;
+					relationId = RangeVarGetRelid(alterTableStmt->relation,
+												  AccessShareLock,
+												  missingOk);
 
 					if (constraint->contype == CONSTR_PRIMARY && constraint->indexname)
 					{
 						char **indexName = &(constraint->indexname);
 						AppendShardIdToName(indexName, shardId);
 					}
 
-					AppendShardIdToName(constraintName, shardId);
+					/* append shardId to constraint names when necessary */
```

Comment:
needs more detailed comment on why we skip certain constraints

Reply:
Also, what is specific about UNIQUE/PRIMARY ?

Reply:
PostgreSQL does not allow two UNIQUE or PRIMARY constraints to have the same name. On the other hand other constraint types can have the same name. (e.g. the `my_seq` CHECK constraint on https://github.com/citusdata/citus/issues/3970 can be created for multiple shards on the same worker node.)

Reply:
PostgreSQL requires constraint names to be unique among the constraints of a single relation or domain. This is enforced by a unique index on conrelid + contypid + conname on pg_constraint. 

However there are two more limitations:
 - UNIQUE constraints create UNIQUE indexes in the schema, and indexes should have unique names. That's the reason why it is not allowed to have 2 UNIQUE constraints on different tables with the same name.
 - Similarly FOREIGN KEY constraints also create indexes that need unique names on the schema.

Reply:
Update: I simplified the logic to consider fixing only  CHECK constraint names. This is consistent with `MergeConstraintsIntoExisting(...)` function in PostgreSQL. `child table is missing constraint` is thrown only when there is a missing check constraint in a child that shares the same name with the parent.

https://github.com/postgres/postgres/blob/6f5c8a8ec23f8ab00da4d2b77bfc8af2a578c4d3/src/backend/commands/tablecmds.c#L14072-L14081